backend/x86_gas.ml

```diff
@@ -238,26 +239,43 @@ let print_instr b = function
   | SQRTSS (arg1, arg2) -> i2 b "sqrtss" arg1 arg2
   | XORPS (arg1, arg2) -> i2 b "xorps" arg1 arg2
   | ANDPS (arg1, arg2) -> i2 b "andps" arg1 arg2
-  | CMPSS (cmp, arg1, arg2) -> i2 b ("cmp" ^ string_of_float_condition cmp ^ "ss") arg1 arg2
+  | CMPSS (cmp, arg1, arg2) ->
+    i2 b ("cmp" ^ string_of_float_condition cmp ^ "ss") arg1 arg2
   | LZCNT (arg1, arg2) -> i2_s b "lzcnt" arg1 arg2
   | TZCNT (arg1, arg2) -> i2_s b "tzcnt" arg1 arg2
-  | SIMD ({ name = "cmpps"; _ }, Some imm, [| arg1; arg2 |]) ->
-    i2 b ("cmp" ^ string_of_float_condition (float_condition_of_int imm) ^ "ps") arg1 arg2
-  | SIMD ({ name = "cmppd"; _ }, Some imm, [| arg1; arg2 |]) ->
-    i2 b ("cmp" ^ string_of_float_condition (float_condition_of_int imm) ^ "pd") arg1 arg2
-  | SIMD ({ name = "crc32"; _ }, None, [| arg1; arg2 |]) ->
+  (* The assembler won't accept these mnemonics directly. *)
+  | SIMD ({ id = Cmpps; _ }, Some imm, [| arg1; arg2 |]) ->
+    i2 b
+      ("cmp" ^ string_of_float_condition (float_condition_of_int imm) ^ "ps")
+      arg1 arg2
+  | SIMD ({ id = Cmppd; _ }, Some imm, [| arg1; arg2 |]) ->
+    i2 b
+      ("cmp" ^ string_of_float_condition (float_condition_of_int imm) ^ "pd")
+      arg1 arg2
+  | SIMD ({ id = Crc32_r64_r64m64; _ }, None, [| arg1; arg2 |]) ->
     i2 b "crc32q" arg1 arg2
-  | SIMD ({ name = ("pcmpestra"|"pcmpestrc"|"pcmpestro"|"pcmpestrs"|"pcmpestrz"); _ }, Some imm, [| arg1; arg2 |]) ->
+  (* These pseudo-instructions map to the same string-comparison instruction. *)
```

Comment:
I think `X86_*` and `Simd_instrs` should only have actual assembly instructions. Emit should translate the pseudo instructions like `Pcmpestra | Pcmpestrc | Pcmpestro | Pcmpestrs | Pcmpestrz` to the corresponding instruction sequences.